setup.py

```diff
@@ -1,6 +1,45 @@
 #!/usr/bin/env python
 
-from setuptools import setup, find_packages
+import glob
+from setuptools import setup
+from setuptools import find_packages
+from setuptools.extension import Extension
+from setuptools.command.sdist import sdist as _sdist
+
+cmdclass = {}
+ext_modules = []
+
+# If cython is available we will re-build our C extensions, otherwise we just
+# compile the packaged C files.
+try:
+    import Cython.Distutils
+    # cython didn't support "from __future__" statements until 0.17.
+    if tuple(int(v) for v in Cython.__version__.split('.')) > (0, 17, 0):
+        extension_filetype = '.pyx'
+        cmdclass.update(build_ext=Cython.Distutils.build_ext)
+except ImportError:
+    extension_filetype = '.c'
+
```

Comment:
Personally, I would list the cythonized modules explicitly. Avoids confusion or unexpected behavior later. But I don't feel strongly about this.